### DISCUSSION TITLE
Old NET Compatability

### DISCUSSION THREAD

**Discussion by smaudet** [snoopwpf/snoopwpf#458] [General]
As of version 5.0.0 support for old .net frameworks was dropped. I'm trying to debug something in MSI land so I can't upgrade. I need the older framework support I think, newer versions of snoop don't even work on this...

I'm currently trying to re-build a 5.0.0 version that is based off of the same stuff as 4.0.1, but with all "modernization" reverted. Are there any other efforts to do this?

**Comment by batzen** [comment]
Those older framework versions are out of support by Microsoft itself.
As .NET 4.6.2 is the oldest supported framework and even Windows 10 came with .NET 4.8 pre-installed, which Windows version are you using?
Any Windows version prior to Windows 10 is also out of support.

**Reply by smaudet** [reply]
I can assure you the "older framework version" is not out of support by Microsoft, as it is whatever is provided by the system (in an msi). I think 4.7.2, even.

At any rate, the code had a bunch of features in it that weren't backwards compatible. I did manage to re-apply many of the fixes without the code "upgrades".

I don't think you should need to explicitly support old versions, but its a bit rude to just do what you feel like and screw other people over.

**Reply by batzen** [reply]
If you are using 4.7.2 there shouldn't be any problems as the lowest supported version is 4.6.2.

It's in no way rude to drop unsupported framework versions.
The issue you reported for your MSI problem is already fixed and was in no way related to framework versions.What doesn't work because of dropped framework version support?

**Reply by smaudet** [reply]
You need to fix your understanding of .net versions then, "framework" != language version and you are using unsupported language versions that do not, in fact, work with old frameworks even if the .net standard claims IL compatability with them.

So it is not possible to *compile* with e.g. net 4.7.2 and yes I know how to install multiple framework versions thanks, it is trying to navigate convincing build tools to use the right version of everything (sometimes not feasible) that makes this decision rather rude, or at least, inconaiderate.

E.g., do not use new style "package" imports, nor new style fancy switch statements, those caused me some grief trying to debug, just because it looks "cleaner" to you.

I.e., it caused me real problems and is mostly just your personal decision, hence why I say rude or inconsiderate, unless you are actually just dropping framework altogether...

**Reply by batzen** [reply]
My understanding of framework, language and SDK versions is excellent.
You should definitely fix your understanding of those things.

It IS possible to compile with .NET 4.7.2.

Why shouldn't i use SDK style projects? Why shouldn't i use modern compiler features that make code understanding and maintenance easier?

You still haven't articulated a single problem this caused you.
Any build errors you could show?
